### PR TITLE
Problem: Missing query denom by name

### LINKED ISSUE #563: Problem: missing query for denom-names
denom-name is not subordinate key of denomid.
denomid ,
denomname,
are both primarykey, indenpendent

so add these commands)
chain-maind query nft denomname   <- query unique denomname
chain-maind query nft denomnames  <- query all denomnames with pagination

currently issuing
mydenomid, mydenomname
mydenomid2, mydenomname

issuing nft with above pattern is not possible.

so further nft handling can cause mistakes

```
{"height":"26","txhash":"415811AB980342625E4D315879F0F76BEE035C215E3CB5C01BFF9FB403B06052","codespace":"nft","code":9,"data":"","raw_log":"failed to execute message; message index: 0: denomName mydenomname has already exists: invalid denom","logs":[],"info":"","gas_wanted":"200000","gas_used":"58907","tx":null,"timestamp":""}
```

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Add endpoint to query denom by name. Fixes #563.

### CODE REVIEW

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Review by JayT106 — APPROVED** [review]
LGTM

File: x/nft/client/rest/query.go [hunk 1]
```diff
@@ -180,6 +183,40 @@ func queryDenom(cliCtx client.Context, queryRoute string) http.HandlerFunc {
...
+			return
+		}
+
+		denomName := mux.Vars(r)[RestParamDenomName]
+		if err := types.ValidateDenomName(denomName); err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+		}
+
+		params := types.NewQueryDenomByNameParams(denomName)
+		bz, err := cliCtx.LegacyAmino.MarshalJSON(params)
```
Comment (JayT106) [inline x/nft/client/rest/query.go:201]: A question relates to the codec. Is any point we will use `gogoprotobuf ` instead of `amino`?

Comment (devashishdxt) [reply x/nft/client/rest/query.go:201]: Ideally, clients should directly use gRPC. Rest can be deprecated in the future.

**Pull request merged**